app/api/sections.js

```diff
@@ -0,0 +1,36 @@
+const { Router } = require('express')
+const router = Router()
+
+const course = require('../models/course')
+const db = require('../models/index')
+const { logger } = require('../../lib/logger')
+const courseService = require('../services/course_service')
+const enrollmentService = require('../services/enrollment_service')
+const sectionService = require('../services/section_service')
+const { generateUserAuthToken, requireAuthentication } = require('../../lib/auth')
+const e = require('express')
+
+router.post('/:course_id/sections', requireAuthentication, async function (req, res) {
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+
+    if (req.body.courseId && req.body.number && req.body.joinCode) {
```

Comment:
Number should be an optional parameter of the body here. The section model auto-increments the number within the scope of the course if it's not provided

Reply:
Also, did we want to autogenerate the joinCode instead of requiring it in the request? It's a unique key so letting the user choose it seems unfavorable